src/screens/AddSiteScreen.js

```diff
@@ -1,204 +1,339 @@
+                        <FormInput
+                            placeholder={directions_placeholder}
+                            value={siteDirectionsText}
+                            onChangeText={this.onUpdateSiteDirectionsText}
+                            containerStyle={largeTextInput}
+                            multiline={true}
+                            maxLength={40}
+                            maxHeight={50}
+                            editable={true}
+                        />
+
+                        <FormLabel>{nearest_town}</FormLabel>
+                        <FormInput
+                            placeholder={nearest_town_placeholder}
+                            value={siteNearestTownText}
+                            onChangeText={this.onUpdateSiteNearestTownText}
+                            editable={true}
+                        />
+
+                        <FormLabel>{accessibility}</FormLabel>
+                        <Picker
+                            selectedValue={accessibilityOption}
+                            onValueChange={this.onUpdateAccessibilityOption}
+                        >
+                            {this.accessibilityOptions()}
+                        </Picker>
+
+                        <FormLabel>{facilities}</FormLabel>
+                        <Picker
+                            selectedValue={facilitiesOption}
+                            onValueChange={this.onUpdateFacilitiesOption}
+                        >
+                            {this.facilitiesOptions()}
+                        </Picker>
+
+                        <Button
+                            onPress={() => alert('submitted')}
+                            title="Open modal"
+                            large
+                            rounded={true}
+                            buttonStyle={[buttonStyle, lastButtonStyle]}
+                            icon={{name: 'plus', type: 'font-awesome'}}
+                            title={submit}
+                        >  {submit}
+                        </Button>
+                    </ScrollView>
+                </Modal>
+            </View>
+        );
+    }
+}
+
+const styles = {
+    descriptionStyle: {
+        color: darkBlue,
+        fontSize: 15
+    },
+    sectionStyle: {
+        backgroundColor: grey
+    },
+    buttonStyle: {
+        marginTop: 10,
+        marginBottom: 10,
+        backgroundColor: navyBlue
+    },
+    headerTitle: {
+        flex: 1,
+        marginTop: 20,
+        color: navyBlue,
+        justifyContent: 'center',
+        alignSelf: 'center',
+    },
+    largeTextInput: {
+        height: 100
+    },
+    modalStyle: {
+        padding: 20,
+        paddingTop: 60
+    },
+    lastButtonStyle: {
+        marginBottom: 100
+    },
+    exitOrResetStyle: {
+        paddingLeft: 20,
+        paddingRight: 20,
+        flexDirection: 'row',
+        justifyContent: 'space-between'
+    }
+
+};
+
```

Comment:
Below, we now link our component to the redux flow, grab the pieces of state we want, and the actions are also connected.